cfme/tests/physical_infrastructure/ui/test_physical_server_details_buttons.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+
+pytestmark = [pytest.mark.tier(3), pytest.mark.provider([LenovoProvider], scope="module")]
+
+
+@pytest.fixture(scope="module")
+def physical_server(appliance, provider):
+    # Get and return the first physical server
+    ph_server = appliance.collections.physical_servers.all(provider)[0]
+    return ph_server
+
+
+# Configuration Button
+def test_refresh_relationships(physical_server, provider):
+    last_refresh = provider.last_refresh_date()
+    physical_server.refresh(provider, handle_alert=True)
+    assert last_refresh != provider.last_refresh_date()
+
+
+# Power Button
+def test_power_off(physical_server, provider):
+    flash_messages = physical_server.power_off()
+    assert flash_messages[0].text == 'Requested Server power_off for the selected server'
```

Comment:
**REQUIRED**
Here and further is not the best approach. `physical_server.power_off()` should return nothing. You need to create an appropriate view and get flash message from there:
```python
physical_server.power_off()
view = provider.create_view("some_view")
view.flash.assert_message('Requested Server power_off for the selected server')
```

Reply:
I understand and I agree with you. Thank you so much for help me.